### PR TITLE
[Ready]Fixes imaginary friend bugs, adds some QoL

### LINKED ISSUE #37049: Imaginary friend couldn't hear any speech.
Issue reported from  Round ID: 86224 (/tg/Station Bagil [ENGLISH] [US-EAST] [100% FREE LAG])

Reporting client version: 512

I gave a player an imaginary friend brain trauma through VV. Once they spawned in they couldn't hear speech from the original person, or from anyone else on the station. They had Common when I checked the language menu.

### LINKED ISSUE #38235: Imaginary Friend Issues
[Round ID]: # 88841

[Testmerges]: # #37790: 'Individual Logs now follow the client around automatically' by optimumtact at commit 74f05572a8
#38225: 'Include item overlays in attack animations (shows the proper color of screwdrivers etc)' by vuonojenmustaturska at commit b4d2b8a0e0

[Reproduction]: # 
1. Was unable to hear anything anyone said, even my own master. (Apparently this was already reported in issue #37049)
2. When my master died (suicide) I didn't die with him, and was able to walk far away from his body and only got teleported back to him when i ran off the z-level
3. Walking on mice caused them to squeak, not sure if only i could hear it or not.
4. I was able to open windoors by walking into them (I still couldn't open or close them by clicking on them)
5. I was able to walk through plastic flaps (not sure if this is intended or not)

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
fix: Fixed a few bugs with imaginary friends, including them not being able to hear anyone or surviving past the owner's death.
add: Imaginary friends can now move into the owner's head.
add: Imaginary friends can now choose to become invisible to the owner at will.
/:cl:

Fixes #37049
Fixes #38235

### CODE REVIEW

**Comment by optimumtact** [comment]
` imaginary firends`

**Comment by Anonmare** [comment]
>tfw even your imaginary friend doesn't want to be around you

File: code/datums/brain_damage/imaginary_friend.dm [hunk 1]
```diff
@@ -138,17 +154,68 @@
 	to_chat(owner, "[rendered]")
 	to_chat(src, "[rendered]")
 
+	//speech bubble
+	if(owner.client)
+		var/image/I = image('icons/mob/talk.dmi', src, "default[say_test(message)]", FLY_LAYER)
```
Comment (Cyberboss) [inline code/datums/brain_damage/imaginary_friend.dm]: why not a mutable_appearance?

Comment (XDTM) [reply code/datums/brain_damage/imaginary_friend.dm]: would it work? i probably copied code that predated it

Comment (AnturK) [reply code/datums/brain_damage/imaginary_friend.dm]: Bump, everything else looks good.

Comment (XDTM) [reply code/datums/brain_damage/imaginary_friend.dm]: Damn i always forget when there's a pending request, will fix asap

**Pull request merged**